presidio-analyzer/presidio_analyzer/recognizer_registry_provider.py

```diff
@@ -0,0 +1,156 @@
+import re
+import yaml
+import os
+import logging
+from pathlib import Path
+from typing import Optional, Union, List, Tuple, Dict, Any
+
+from presidio_analyzer import RecognizerRegistry, EntityRecognizer, PatternRecognizer
+from presidio_analyzer.nlp_engine import NlpEngineProvider, NlpEngine
+import presidio_analyzer.recognizer_registry
+
+logger = logging.getLogger("presidio-analyzer")
+
+
+class RecognizerRegistryProvider:
+    """
+    Utility class for loading Recognizer Registry.
+
+    Use this class to load recognzer registry from a yaml file
+    """
+
+    def __init__(self, conf_file: Optional[Union[Path, str]] = None, registry_configuration: Optional[Dict] = None):
+        self.configuration = self._get_configuration(conf_file, registry_configuration)
+        return
+    
+    def _add_missing_keys(self, configuration: Dict, conf_file: Union[Path, str]):
+        defaults = yaml.safe_load(open(conf_file))
+        configuration.update({k: v for k, v in defaults.items() if k not in list(configuration.keys())})
+        return configuration
+
+    def _get_configuration(self, conf_file: Union[Path, str], registry_configuration: Dict) -> Union[Dict[str, Any]]:
+        configuration = {}
+
+        if registry_configuration:
+            configuration = registry_configuration.copy()
+
+        if not conf_file:
+            configuration = self._add_missing_keys(configuration=configuration, conf_file=self._get_full_conf_path())
```

Comment:
Can this raise an `IOError`? If yes, worth wrapping with try/except

Reply:
Doing it underneath. It would happen only if someone would remove the default conf file (we use the same functionality in the nlp engine provider).